demos/digit-prediction/digits_prediction.nlogo

```diff
@@ -0,0 +1,648 @@
+extensions [ py ]
+
+globals [
+  coords-x coords-y color-list digit
+  zero one two three four five six seven eight nine
+]
+
+to setup-py
+  clear-all
+  py:setup py:python
+  (py:run
+    "import numpy as np"
+    "from keras.models import load_model"
+  )
+  ;; load model
+  py:run "model = load_model(\"model.h5\")"
+
+  ;; create coords list, useful to make color_list
+  set coords-x (range max-pxcor (min-pxcor - 1) -1)
+  set coords-y (range max-pycor (min-pycor - 1) -1)
+  set color-list []
+end
+
+to draw
+  ifelse mouse-down?
+  [
+    ask patch mouse-xcor mouse-ycor
+    [
+      ifelse erase?
+      [
+        set pcolor black
+      ]
+      [
+        set pcolor white
+      ]
+    ]
+  ]
+  [
+    predict
+  ]
+  display
+end
+
+to predict
+  set color-list []
+  foreach coords-x
+  [
+    x -> foreach coords-y
+    [
+      y ->
+        let c [ pcolor / 10 ] of patch x y
+        set color-list lput c color-list
+    ]
+  ]
+  py:set "images" color-list
+  py:run "array = np.array(images).reshape(1, 28, 28, 1)"
+  py:run "y_pred = model.predict(array)[0]"
+  set digit py:runresult "np.argmax(y_pred)"
+
+  ;; return probabilities per digit (softmax layer) in %
+  set zero   py:runresult "y_pred[0] * 100"
+  set one    py:runresult "y_pred[1] * 100"
+  set two    py:runresult "y_pred[2] * 100"
+  set three  py:runresult "y_pred[3] * 100"
+  set four   py:runresult "y_pred[4] * 100"
+  set five   py:runresult "y_pred[5] * 100"
+  set six    py:runresult "y_pred[6] * 100"
+  set seven  py:runresult "y_pred[7] * 100"
+  set eight  py:runresult "y_pred[8] * 100"
+  set nine   py:runresult "y_pred[9] * 100"
```

Comment:
I would just do:

```
  set probabilities py:runresult "y_pred"
```

where `probabilities` is a global. Then, in your monitors, you do (for instance) `100 * item 3 probabilities`. You'd also need to zero out the `probabilities` list in `setup-py`: `set probabilities n-values 10 [ 0 ]`.